conformance/utils/kubernetes/helpers.go

```diff
@@ -295,6 +324,49 @@ func GatewayMustHaveCondition(
 	require.NoErrorf(t, waitErr, "error waiting for Gateway status to have a Condition matching expectations")
 }
 
+// ListenerSetMustHaveCondition checks that the supplied ListenerSet has the supplied Condition,
+// halting after the specified timeout is exceeded.
+func ListenerSetMustHaveCondition(
+	t *testing.T,
+	client client.Client,
+	timeoutConfig config.TimeoutConfig,
+	lsNN types.NamespacedName,
+	expectedCondition metav1.Condition,
+) {
+	t.Helper()
+
+	waitErr := wait.PollUntilContextTimeout(
+		context.Background(),
+		1*time.Second,
+		timeoutConfig.GatewayMustHaveCondition,
```

Comment:
you probably want to have its own timeout config for listenerSet?

Reply:
Good idea!